### PR TITLE
parser,ddl: add background job syntax and meta management for resource control

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: ref #44517

Problem Summary:

### What is changed and how it works?

Resource control add new syntax for background tasks management as following:
```sql
// Set background tasks
CREATE RESOURCE GROUP `rg1` ru_per_sec = 1000 BACKGROUND=(TASK_NAMES="br,lightning");
// change settings
ALTER RESOURCE GROUP `rg1` BACKGROUND = (TASK_NAMES="br");
// reset background tasks
ALTER RESOURCE GROUP `rg1` BACKGROUND = NULL;
```

NOTE: in the first phase, we only support changing the `default` resource group's background config.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -1823,15 +1827,55 @@ DirectResourceGroupOption:
...
+	{
+		if !ast.CheckBackgroundAppend($1.([]*ast.ResourceGroupBackgroundOption), $3.(*ast.ResourceGroupBackgroundOption)) {
+			yylex.AppendError(yylex.Errorf("Dupliated background options specified"))
+			return 1
+		}
+		$$ = append($1.([]*ast.ResourceGroupBackgroundOption), $3.(*ast.ResourceGroupBackgroundOption))
+	}
+
+DirectResourceGroupBackgroundOption:
+	"TASK_NAMES" EqOpt stringLit
```
Comment (CabinfeverB) [inline parser/parser.y:1876]: Is the task name an enumeration type? I find PD and TiDB won't check whether name is valid, user can't get feedback if the name is wrong.

Comment (HuSharp) [reply parser/parser.y:1876]: Yes, you can check the enum here.
https://github.com/tikv/client-go/pull/868/files#diff-6bd1a8e867c588123e135e97a8474639b648d9b6e4ab28c5b29338b388505dfbR41

Comment (CabinfeverB) [reply parser/parser.y:1876]: My main concern is that when setting the backgroup, if the user input is wrong, it will not report an error.

Comment (glorv) [reply parser/parser.y:1876]: Added a check in ddl api

**Comment by glorv** [comment]
@CabinfeverB @HuSharp @nolouch  PTAL again, thanks

**Review by nolouch — APPROVED** [review]
lgtm

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -3189,12 +3192,24 @@ func SetDirectResourceGroupSettings(resourceGroupSettings *model.ResourceGroupSe
...
-		for _, opt := range opt.ResourceGroupRunawayOptionList {
-			err := SetDirectResourceGroupRunawayOption(resourceGroupSettings, opt.Tp, opt.StrValue, opt.IntValue)
-			if err != nil {
+		for _, opt := range opt.RunawayOptionList {
+			if err := SetDirectResourceGroupRunawayOption(resourceGroupSettings, opt.Tp, opt.StrValue, opt.IntValue); err != nil {
+				return err
+			}
+		}
+	case ast.ResourceGroupBackground:
+		if groupInfo.Name.L != rg.DefaultResourceGroupName {
```
Comment (HuSharp) [inline ddl/ddl_api.go:3204]: I noticed this restriction, so maybe need to update pr description?

Comment (glorv) [reply ddl/ddl_api.go:3204]: There is a `NOTE` in the PR description.

Comment (HuSharp) [reply ddl/ddl_api.go:3204]: oops, sorrrry! I must be blind!

File: parser/ast/ddl.go [hunk 3]
```diff
@@ -2138,59 +2147,73 @@ const (
...
+		ctx.WritePlain("BACKGROUND ")
+		ctx.WritePlain("= ")
+		if len(n.BackgroundOptions) > 0 {
+			ctx.WritePlain("(")
+			for i, option := range n.BackgroundOptions {
+				if i > 0 {
+					ctx.WritePlain(", ")
+				}
+				if err := option.Restore(ctx); err != nil {
+					return errors.Annotatef(err, "An error occurred while splicing ResourceGroup Background Option")
```
Comment (HuSharp) [inline parser/ast/ddl.go:2206]: Do we need to unify error with `ResourceGroupRunaway` to expose `i`?
```suggestion
					return errors.Annotatef(err, "An error occurred while splicing ResourceGroup Background Option: [%v]", i")
```

File: parser/ast/ddl.go [hunk 4]
```diff
@@ -2138,59 +2147,73 @@ const (
...
+		ctx.WritePlain("QUERY_LIMIT ")
+		ctx.WritePlain("= ")
+		if len(n.RunawayOptionList) > 0 {
+			ctx.WritePlain("(")
+			for i, option := range n.RunawayOptionList {
+				if i > 0 {
+					ctx.WritePlain(" ")
+				}
+				if err := option.Restore(ctx); err != nil {
+					return errors.Annotatef(err, "An error occurred while splicing CreateResourceGroupStmt Option: [%v]", i)
```
Comment (HuSharp) [inline parser/ast/ddl.go:2189]:
```suggestion
					return errors.Annotatef(err, "An error occurred while splicing ResourceGroupRunaway Option: [%v]", i)
```

**Review by HuSharp — APPROVED** [review]
merge master failed, reset LGTM!

**Review by 3pointer — APPROVED** [review]

**Pull request merged**